mesonbuild/interpreter.py

```diff
@@ -2319,7 +2334,12 @@ def func_include_directories(self, node, args, kwargs):
         is_system = kwargs.get('is_system', False)
         if not isinstance(is_system, bool):
             raise InvalidArguments('Is_system must be boolean.')
-        i = IncludeDirsHolder(build.IncludeDirs(self.subdir, args, is_system))
+        for n, a_is_system in incdirs_postcheck:
+            if a_is_system != is_system:
+                raise InvalidArguments(
+                        'Unsupported mixing of system and non-system headers: argument at position {} contains {} headers.'.format(
```

Comment:
__[flake8]__
* [E126] continuation line over-indented for hanging indent

<!-- Comment by [SideCI](https://sideci.com) -->